### PR TITLE
Implement legacy secondaries

### PR DESCRIPTION
**Pull request by OYTIS**
Common functionality of legacy and virtual secondaries is isolated into ManagedSecondary class

### CODE REVIEW

File: src/utils.cc [hunk 1]
```diff
@@ -270,6 +271,27 @@ void Utils::copyDir(const boost::filesystem::path &from, const boost::filesystem
   }
 }
```
Comment (patriotyk) [inline src/utils.cc]: I added the similar method in my new PR )))

File: src/uptane/legacysecondary.cc [hunk 2]
```diff
@@ -7,70 +7,34 @@
...
-      }
-    }
-  }
-  detected_attack = "";
-  Utils::writeFile(sconfig.target_name_path.string(), content);
-  return true;
+  std::string output
+  int rs = Utils::shell(sconfig.flasher.string() + " --hardware-identifier " + sconfig.ecu_hardware_id + " --ecu-identifier " + sconfig.ecu_serial + " --firmware " + sconfig.firmware_path.string(), &output);
+
+  if(rs != 0)
```
Comment (patriotyk) [inline src/uptane/legacysecondary.cc]: It is better always use {}. If someone will need to add some code inside if statement, he could not notice that there is no {} symbols.

File: src/uptane/legacysecondary.h [hunk 3]
```diff
@@ -1,33 +1,21 @@
-#ifndef UPTANE_LEGACYSECONDARY_H_
-#define UPTANE_LEGACYSECONDARY_H_
+#ifndef UPTANE_VIRTUALSECONDARY_H_
+#define UPTANE_VIRTUALSECONDARY_H_
```
Comment (patrickvacek) [inline src/uptane/legacysecondary.h]: That seems like a typo, right? It still should be `UPTANE_LEGACYSECONDARY_H_`.

**Comment by patrickvacek** [comment]
I fixed the include guard issue and merged. While I generally agree with Serhiy about preferring brackets around single-clause branches, I don't think it is strictly necessary nor an impediment to moving forward with the merge and further development. (Nor is it a strict requirement in the Google style guide.) But perhaps before this is all finally merged into master we can look more closely at things like that.

**Pull request merged**